samples/directx/d3d10_opencl.cpp

```diff
@@ -0,0 +1,397 @@
+#define WIN32_LEAN_AND_MEAN
+#include <windows.h>
+#include <d3d10.h>
+
+#include <string>
+#include <iostream>
+#include <queue>
+
+#include "opencv2/core.hpp"
+#include "opencv2/core/directx.hpp"
+#include "opencv2/core/ocl.hpp"
+#include "opencv2/imgproc.hpp"
+#include "opencv2/videoio.hpp"
+#include "winapp.hpp"
+
+#pragma comment (lib, "d3d10.lib")
+
+
+class D3D10WinApp : public WinApp
+{
+public:
+    D3D10WinApp(int width, int height, std::string& window_name, cv::VideoCapture& cap) :
+        WinApp(width, height, window_name)
+    {
+        m_shutdown = false;
+        m_mode = 0;
+        m_modeStr[0] = cv::String("No processing");
+        m_modeStr[1] = cv::String("Processing on CPU");
+        m_modeStr[2] = cv::String("Processing on GPU");
+        m_disableProcessing = false;
+        m_cap = cap;
+    }
+
+    ~D3D10WinApp() {}
+
+    int onClose(void)
+    {
+        m_shutdown = true;
+        cleanup();
+        ::DestroyWindow(m_hWnd);
```

Comment:
`DestroyWindow` call (or almost all `onClose` method) should be moved into `WinApp` class (because window management can be there).